core/network/src/main/kotlin/com/google/samples/apps/nowinandroid/core/network/demo/DemoNiaNetworkDataSource.kt

```diff
@@ -42,13 +43,27 @@ class DemoNiaNetworkDataSource @Inject constructor(
     @OptIn(ExperimentalSerializationApi::class)
     override suspend fun getTopics(ids: List<String>?): List<NetworkTopic> =
         withContext(ioDispatcher) {
-            assets.open(TOPICS_ASSET).use(networkJson::decodeFromStream)
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                assets.open(TOPICS_ASSET).use(networkJson::decodeFromStream)
+            } else {
+                // Use decodeFromString to capability with API 24 below.
+                // https://github.com/Kotlin/kotlinx.serialization/issues/2457#issuecomment-1786923342
+                val topicsJsonString = assets.readText(TOPICS_ASSET)
+                networkJson.decodeFromString(topicsJsonString)
+            }
```

Comment:
```suggestion
            assets.open(TOPICS_ASSET).use {
                if (SDK_INT >= N) networkJson.decodeFromStream(it)
                // https://github.com/Kotlin/kotlinx.serialization/issues/2457#issuecomment-1786923342
                else it.bufferedReader().use(BufferedReader::readText).let(networkJson::decodeFromString)
            }
```

Reply:
Thank you. I accept your suggestion but not totally same. I made a private function with generic type for two method, `getTopics` and `getNewsResources`.